monitor.cpp

```diff
@@ -3,36 +3,54 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
+#include <functional>
+#include <vector>
+
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+using std::function;
+using std::string;
+using std::vector;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
+void displayCriticalAlert(const std::string& message) {
+    cout << message << "\n";
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        cout << "\r* " << flush;
+        sleep_for(seconds(1));
+        cout << "\r *" << flush;
+        sleep_for(seconds(1));
     }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+}
+
+bool isTemperatureCritical(float temperature) {
+    return temperature > 102 || temperature < 95;
```

Comment:
You can further abstract this by defining a function like `isParameterOutOfRange(float inputValue, float lowerLimit, float upperLimit)` and use it in each of the three functions. Advantage may be: We need to test the range-functionality only once. Even when we add more parameters to the report, we don't need to add three tests for each parameters.